src/extension/serializer.ts

```diff
@@ -512,8 +515,45 @@ export class GrpcSerializer extends SerializerBase {
     )
   }
 
-  private async initParserClient(transport?: GrpcTransport) {
-    this.client = initParserClient(transport ?? (await this.server.transport()))
+  private async initClient(): Promise<void> {
```

Comment:
I'd likely be cleaner/easier to have the `KernelServer` supply transports for both GRPC and ES.